data/synth_timing.xdc

```diff
@@ -292,20 +296,58 @@ set usb_conf_names {usrusb_softcn usrusb_sus usrusb_spd}
 set_output_delay -clock vclk_extusb -max 0 [get_ports $usb_conf_names]
 set_output_delay -clock vclk_extusb -min 0 [get_ports $usb_conf_names]
 
-## PMOD 0 - asynchronous GPIO
+## PMOD 0
+#   SPI (or I^2C or UART or GPIO, but let us assume SPI is most critical)
+# Use *similar* methodology as Arduino Shield SPI, but with less I/O delay on
+# some shared ports to avoid timing violations on I^2C and UART paths.
+# Distance to PMOD 0 SPI pins = ~20-25 mm  x2(clk there + data back)
+set pmod0_spi_trce_dly_min [expr {20 * 2 * 0.004}]
+set_input_delay -clock clk_pmod0_spi -max [expr {$sclk_ns - 22                         }] [get_ports {pmod0[2]}] ;# CIPO
+set_input_delay -clock clk_pmod0_spi -min [expr {$sclk_ns/2.0 + $pmod0_spi_trce_dly_min}] [get_ports {pmod0[2]}]
+set_output_delay -clock clk_sys -max [expr {$clk_sys_ns - 14}] [get_ports {pmod0[3]}] ;# SCLK
+set_output_delay -clock clk_sys -min 0                         [get_ports {pmod0[3]}]
+set_output_delay -clock clk_pmod0_spi -max [expr {$sclk_ns - 6           }] [get_ports {pmod0[1]}] ;# COPI
+set_output_delay -clock clk_pmod0_spi -min [expr {           $sclk_ns/2.0}] [get_ports {pmod0[1]}]
+set_output_delay -clock clk_pmod0_spi -max [expr {$sclk_ns - 6           }] [get_ports {pmod0[0] pmod0[6] pmod0[7]}] ;# chip selects
+set_output_delay -clock clk_pmod0_spi -min [expr {           $sclk_ns/2.0}] [get_ports {pmod0[0] pmod0[6] pmod0[7]}]
+#   GPIO/PWM (that are not SPI)
 # Could be human-scale or processor-scale (bit-bashing) time periods.
 # Put a multicycle constraint on them both in the timing exceptions section,
 # and zero I/O delays here.
-set_input_delay -clock vclk_sys -max 0 [get_ports {pmod0[*]}]
-set_input_delay -clock vclk_sys -min 0 [get_ports {pmod0[*]}]
-set_output_delay -clock vclk_sys -max 0 [get_ports {pmod0[*]}]
-set_output_delay -clock vclk_sys -min 0 [get_ports {pmod0[*]}]
-## PMOD 1 - asynchronous GPIO
-# Use same methodology as PMOD 0
-set_input_delay -clock vclk_sys -max 0 [get_ports {pmod1[*]}]
-set_input_delay -clock vclk_sys -min 0 [get_ports {pmod1[*]}]
-set_output_delay -clock vclk_sys -max 0 [get_ports {pmod1[*]}]
-set_output_delay -clock vclk_sys -min 0 [get_ports {pmod1[*]}]
+set pmod0_gpio_names {pmod0[4] pmod0[5]}
+set_input_delay -clock vclk_sys -max 0 [get_ports $pmod0_gpio_names]
+set_input_delay -clock vclk_sys -min 0 [get_ports $pmod0_gpio_names]
+set_output_delay -clock vclk_sys -max 0 [get_ports $pmod0_gpio_names]
+set_output_delay -clock vclk_sys -min 0 [get_ports $pmod0_gpio_names]
+
+## PMOD 1
+#   SPI (or I^2C or UART or GPIO, but let us assume SPI is most critical)
+# Use *similar* methodology as Arduino Shield SPI, but with less I/O delay on
+# some shared ports to avoid timing violations on I^2C and UART paths.
+# Distance to PMOD 1 SPI pins = ~25-30 mm  x2(clk there + data back)
```

Comment:
Is this larger than the 20 mentioned above for pmod0?

Reply:
Ah, thanks for catching this. I updated the comment but not the code!

Reply:
Resolved now.